http-native/src/main/java/org/ballerinalang/net/http/HttpCallableUnitCallback.java

```diff
@@ -47,6 +51,11 @@
     public void notifySuccess(Object result) {
         if (result == null) { // handles nil return and end of resource exec
             requestMessage.waitAndReleaseAllEntities();
+            ObserverContext observerContext
```

Comment:
Shouldn't we check whether `ObserveUtils.isObservabilityEnabled()` before looking for the `ObserverContext`?

Reply:
`ObserverContext` will be null when observability is disabled hence that check is redundant. Should I add it anyway?

Reply:
Yeah, we can get rid of the map lookup and the cast if check it right? Shall we do that by considering the potential perf gain?

Reply:
Fixed